src/gobimport/reader.py

```diff
@@ -43,21 +43,35 @@ def __init__(self, source, app, dataset):
         self.source = source
         self.app = app
 
-        secure_types = [f"GOB.{type.name}" for type in GOB_SECURE_TYPES]
+        self.secure_types = [f"GOB.{type.name}" for type in GOB_SECURE_TYPES]
         mapping = dataset["gob_mapping"]
 
         catalogue = dataset['catalogue']
         entity = dataset['entity']
         gob_attributes = GOBModel().get_collection(catalogue, entity)["all_fields"]
 
         self.secure_attributes = []
-        for maps_on, map_spec in mapping.items():
-            gob_attr = gob_attributes[maps_on]
-            if gob_attr["type"] in secure_types:
-                self.secure_attributes.append(map_spec["source_mapping"])
+        self.set_secure_attributes(mapping, gob_attributes)
 
         self._connection = None
 
+    def set_secure_attributes(self, mapping, gob_attributes):
```

Comment:
Please add unit-test(s) for this one. Maybe it will be good to make get_secure_attributes and let it return secure_attributes. You can also pass secure_types there and make this a class method. I guess it will easy to test then.

Reply:
Unit tests added